waku/v2/protocol/relay/waku_relay.go

```diff
@@ -280,12 +280,20 @@ func (w *WakuRelay) Publish(ctx context.Context, message *pb.WakuMessage, opts .
 		if err != nil {
 			return pb.MessageHash{}, err
 		}
+		_, err = w.subscribeToPubsubTopic(params.pubsubTopic)
```

Comment:
why subscribe only in case pubsubtopic is empty?
shouldn't we return errro in this case as well simliar to line 294

Reply:
For relay REST API, it supports publish a message without pubsubTopic, this is a quick fix for it.
Not sure if there's a better way for this scenario.